### DISCUSSION TITLE
Recommended usage of Footer and Identifier

### DISCUSSION THREAD

**Discussion by paulathevalley** [uswds/uswds#4625] [Q&A] [answered]
👋 What is the recommended implementation for a footer element that's visually styled to match the rest of the page and the Identifier component exactly as it looks on USWDS? 

I'm curious about the HTML structure specifically.

Other websites seem to vary in their approach, so I'm wondering about the recommended usage, especially given the best practice that [all page content must be contained by landmarks](https://dequeuniversity.com/rules/axe/3.5/region).

## Research of existing implementation:

1. Identifier component is a sibling to `<footer>` element:
  ```html
  <footer> ... </footer>
  <div class="usa-identifier"> ... </div> <!-- should this have `role="contentinfo"`? -->
  ```
  - https://digital.gov
  - https://18f.gsa.gov

2. Identifier component is a descendant of `<footer>` element:
```html
<footer> 
   ...
  <div class="usa-identifier"> ... </div>
</footer>
```
- https://search.gov
- https://login.gov

thank you 🙏

**Comment by mejiaj** [comment, marked as answer]
Hey @paulathevalley! The original intention was to have it as a sibling of footer. On the [design system website](https://designsystem.digital.gov/), it's also as a sibling.

The question of Identifier requiring a landmark role is interesting. I don't think we're opposed to it, but at the time of our testing it never seemed to be an issue.

We're open to any suggestions to improve it's accessibility!

**Reply by paulathevalley** [reply]
@mejiaj Thank you James! 

If you have already done extensive accessibility testing on it without problems, then my instinct is to change nothing. 👍

**Reply by jkjustjoshing** [reply]
Thanks @mejiaj
You said that no issues came up when testing. For my own curiosity, was the focus of the testing explicitly about the footer+identifier setup, or was it more general testing? I’m just trying to understand how to think about the tradeoffs between implementing the footer as it exists on [designsystem.digital.gov](http://designsystem.digital.gov/), versus adhering to Deque’s landmark recommendation.

**Reply by mejiaj** [reply]
Hey @jkjustjoshing, with layout it was more general testing.

**Comment by ryparker** [comment]
To keep both sections inside a page landmark, place the footer content and Identifier inside one top-level `footer`:

```html
<footer>
  <div class="usa-footer">
    <!-- Footer content -->
  </div>
  <div class="usa-identifier">
    <!-- Identifier content -->
  </div>
</footer>
```

Use the appropriate footer variant class on the inner wrapper when needed. That lets the two sections retain their styling while sharing the page's `contentinfo` landmark. Do not put another `footer` element inside the wrapper.

A top-level HTML `footer` already supplies the landmark, so the Identifier does not need a separate `role="contentinfo"`. This follows the [ARIA landmark guidance](https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/contentinfo.html), which recommends one page-level contentinfo landmark.

The earlier reply describes general layout testing. It should not be taken as a specific finding that leaving the Identifier outside every landmark is equivalent. Test the combined layout and landmark navigation in the implementation as well.
